src/node/operator_executors/sink_executor.rs

```diff
@@ -0,0 +1,350 @@
+        helper.synchronize().await;
+
+        // Execute the `run` method.
+        let mut read_stream: ReadStream<T> = self.read_stream.take().unwrap();
+        slog::debug!(
+            crate::TERMINAL_LOGGER,
+            "Node {}: Running Operator {}",
+            self.config.node_id,
+            self.config.get_name(),
+        );
+
+        tokio::task::block_in_place(|| {
+            self.executor.execute_run(&mut read_stream);
+        });
+
+        // Run the `setup` method.
+        let setup_context = OneInOneOutSetupContext::new(read_stream.id());
+        // TODO (Sukrit): Implement deadlines and `setup` method for the `Sink` operators.
+
+        // Process messages on the incoming stream.
+        let process_stream_fut = helper.process_stream(
+            read_stream,
+            &mut (*self.executor),
+            &channel_to_event_runners,
+            &setup_context,
+        );
+
+        // Shutdown and cleanup.
+        loop {
+            tokio::select! {
+                _ = process_stream_fut => break,
+                notification_result = channel_from_worker.recv() => {
+                    match notification_result {
+                        Ok(notification) => {
+                            match notification {
+                                OperatorExecutorNotification::Shutdown => { break; }
+                            }
+                        }
+                        Err(e) => {
+                            slog::error!(
+                                crate::TERMINAL_LOGGER,
+                                "Sink Executor {}: error receiving notifications {:?}",
+                                self.operator_id(),
+                                e,
+                                );
+                            break;
+                        }
+                    }
+                }
+            }
+        }
+
+        // Invoke the `destroy` method.
+        tokio::task::block_in_place(|| self.executor.execute_destroy());
+
+        channel_to_worker
+            .send(WorkerNotification::DestroyedOperator(self.operator_id()))
+            .unwrap();
+    }
+}
+
+impl<T> OperatorExecutorT for SinkExecutor<T>
+where
+    T: Data + for<'b> Deserialize<'b>,
+{
+    fn execute<'a>(
+        &'a mut self,
+        channel_from_worker: broadcast::Receiver<OperatorExecutorNotification>,
+        channel_to_worker: mpsc::UnboundedSender<WorkerNotification>,
+        channel_to_event_runners: broadcast::Sender<EventNotification>,
+    ) -> Pin<Box<dyn Future<Output = ()> + 'a + Send>> {
+        Box::pin(self.execute(
+            channel_from_worker,
+            channel_to_worker,
+            channel_to_event_runners,
+        ))
+    }
+
+    fn lattice(&self) -> Arc<ExecutionLattice> {
+        Arc::clone(&self.helper.as_ref().unwrap().lattice)
+    }
+
+    fn operator_id(&self) -> OperatorId {
+        self.config.id
+    }
+}
+
+pub struct ReadOnlySinkMessageProcessor<O, S, T, U, V>
```

Comment:
I assume this is not quite ready because this processor is not used anywhere.

Reply:
This was removed in the final commits.